### PR TITLE
fix(platform): dynamic form item should support select controlTemplate to improve two-column rendering

### PR DESCRIPTION
**Pull request by littleamigo**
fixes https://github.com/SAP/fundamental-ngx/issues/9178

Adds some properties from select to dynamic form item, allowing further customization of select in form generator/wizard generator

### CODE REVIEW

File: libs/platform/src/lib/form/form-generator/interfaces/dynamic-form-item.ts
Comment (N1XUS) [inline libs/platform/src/lib/form/form-generator/interfaces/dynamic-form-item.ts:242]: - [ ] TemplateRef is not imported;
- [ ] Property should be optional

**Review by N1XUS — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/form/form-generator/interfaces/dynamic-form-item.ts
Comment (g-cheishvili) [reply libs/platform/src/lib/form/form-generator/interfaces/dynamic-form-item.ts:242]: I do not like this. This interface is intended for use in every control, if you enhance select to use this, then other provided controls should use this as well.
For achieving custom stuff, I would go for [this approach](https://sap.github.io/fundamental-ngx/#/platform/form-generator#custom)

Comment (littleamigo) [reply libs/platform/src/lib/form/form-generator/interfaces/dynamic-form-item.ts]: I do not understand your proposal, I'm afraid. Select is not a custom control act. https://sap.github.io/fundamental-ngx/#/platform/wizard-generator > Wizard Generator Elements: select.
In addition, already 3 properties from select were added: "secondaryKey, showSecondaryText, secondaryTextAlignment" recently by mikerodonnell89 and he asked me to add this one myself as we figured this was missing in the first place.

Comment (g-cheishvili) [reply libs/platform/src/lib/form/form-generator/interfaces/dynamic-form-item.ts]: @littleamigo I created issue, which describes this. https://github.com/SAP/fundamental-ngx/issues/9349

It would be nice if we made this happen this way

**Review by platon-rov — APPROVED** [review]
As agreed here https://github.com/SAP/fundamental-ngx/issues/9349 we can merge this one to unblock customers and then think about refactoring.

**Comment by g-cheishvili** [comment]
> As agreed here #9349 we can merge this one to unblock customers and then think about refactoring.

Yes, but without `TemplateRef` import and optional, this will not be buildable and we can not merge it

**Comment by mikerodonnell89** [comment]
> > As agreed here #9349 we can merge this one to unblock customers and then think about refactoring.
> 
> Yes, but without `TemplateRef` import and optional, this will not be buildable and we can not merge it

Looks like these have already been addressed

**Comment by littleamigo** [comment]
Let me know if there is anything missing by me!

I'd also accept a better default body template implementation for the 2-column select which renders the 2 values with a separating colon, for example. The default implementation is certainly for 99% of the use cases inappropriate:
image
image

**Comment by littleamigo** [comment]
@g-cheishvili @mikerodonnell89 @N1XUS @platon-rov How are we going to merge this thing?

**Comment by littleamigo** [comment]
it look like commit messages have to obey specific rules? why is that? looks strange to me because i'm used to have as many commit as i want in the way i want to achieve the goal of my pull request. i typically squash and merge them at the end. then, only the pull request title goes into the single commit message. this is the one worth to be checked for specific rules FMPOV.

**Comment by littleamigo** [comment]
> > it looks like commit messages have to obey specific rules? why is that? looks strange to me because i'm used to have as many commit as i want in the way i want to achieve the goal of my pull request. i typically squash and merge them at the end. then, only the pull request title goes into the single commit message. this is the one worth to be checked for specific rules FMPOV.
> 
> Could you just fix the last commit and rebase on main? What you're saying makes sense and I think it is the better approach, would you mind creating an issue for that?

@g-cheishvili not sure what exactly you want me to do:
- rebase on main -> no problem, i can do
- fix the last commit -> what is wrong with the last [commit](https://github.com/SAP/fundamental-ngx/pull/9326/commits/0c52b48ab14ac77d4ec0cf8ff6c24e1fecd3db08)? or do you want me to revert the last 2 commits because they are not required?

**Comment by g-cheishvili** [comment]
> > > it looks like commit messages have to obey specific rules? why is that? looks strange to me because i'm used to have as many commit as i want in the way i want to achieve the goal of my pull request. i typically squash and merge them at the end. then, only the pull request title goes into the single commit message. this is the one worth to be checked for specific rules FMPOV.
> > 
> > 
> > Could you just fix the last commit and rebase on main? What you're saying makes sense and I think it is the better approach, would you mind creating an issue for that?
> 
> @g-cheishvili not sure what exactly you want me to do:
> 
> * rebase on main -> no problem, i can do
> * fix the last commit -> what is wrong with the last [commit](https://github.com/SAP/fundamental-ngx/pull/9326/commits/0c52b48ab14ac77d4ec0cf8ff6c24e1fecd3db08)? or do you want me to revert the last 2 commits because they are not required?

just fix the name to be compliant with conventional commits, that's all

**Comment by N1XUS** [comment]
Closing in favour of #9497

**Comment by littleamigo** [comment]
N1XUS confirmed he did some refactorings to the form generator and applied my proposed changes to it

**Pull request closed without merging**
